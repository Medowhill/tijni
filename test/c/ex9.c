#include <stdlib.h>

long f(int *p, int x) {
  *p = x;
  return (long) malloc(sizeof(int));
}

void g(int *p, int x) {
  int *q = (int *) f(p, x);
  *q = x;
}
