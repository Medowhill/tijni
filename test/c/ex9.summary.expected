{
  "f": {
    "params": [ { "name": "%p(f)", "used": true, "freed": false }, { "name": "%x(f)", "used": false, "freed": false } ],
    "ret": [ "Allocsite(0)" ],
    "env": {
      "%p(f)": [ "'%p(f)" ],
      "%x(f)": [ "'%x(f)" ],
      "%call(f)": [ "Allocsite(0)" ],
      "%0(f)": [ "Allocsite(0)" ],
      "'%p(f)": [ "'%x(f)" ]
    }
  },
  "g": {
    "params": [ { "name": "%p(g)", "used": true, "freed": false }, { "name": "%x(g)", "used": false, "freed": false } ],
    "ret": [],
    "env": {
      "%p(f)": [ "'%p(g)" ],
      "%x(f)": [ "'%x(g)" ],
      "%call(f)": [ "Allocsite(0)" ],
      "%0(f)": [ "Allocsite(0)" ],
      "%p(g)": [ "'%p(g)" ],
      "%x(g)": [ "'%x(g)" ],
      "%call(g)": [ "Allocsite(0)" ],
      "%0(g)": [ "Allocsite(0)" ],
      "'%p(g)": [ "'%x(g)" ],
      "Allocsite(0)": [ "'%x(g)" ]
    }
  }
}